Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift

```diff
@@ -0,0 +1,298 @@
+                                await updateSceneView(for: location, heading: heading)
+                            }
+                        }
+                    }
+                    group.addTask {
+                        for await heading in locationDataSource.headings {
+                            self.currentHeading = heading
+                            if let location = currentLocation {
+                                await updateSceneView(for: location, heading: heading)
+                            }
+                        }
+                    }
+                }
+            } catch {}
+        }
+        .toolbar {
+            ToolbarItem(placement: .bottomBar) {
+                if !calibrationViewIsHidden {
+                    calibrationView
+                }
+            }
+        }
+        .overlay(alignment: .top) {
+            accuracyView
+                .multilineTextAlignment(.center)
+                .frame(maxWidth: .infinity, alignment: .center)
+                .padding(8)
+                .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+        }
+    }
+    
+    /// If necessary, updates the scene view's camera controller for a new location coming
+    /// from the location datasource.
+    @MainActor
+    private func updateSceneView(for location: Location, heading: Double) {
+        // Do not use cached location more than 10 seconds old.
+        guard abs(lastLocationTimestamp?.timeIntervalSinceNow ?? 0) < 10 else { return }
+        
+        // Make sure that horizontal and vertical accuracy are valid.
+        guard location.horizontalAccuracy > validAccuracyThreshold,
+              location.verticalAccuracy > validAccuracyThreshold else { return }
+        
+        // Make sure either the initial camera is not set, or we need to update the camera.
+        guard !initialCameraIsSet || shouldUpdateCamera(for: location) else { return }
+        
+        // Add some of the vertical accuracy to the z value of the position, that way if the
+        // GPS location is not accurate, we won't end up below the earth's surface.
+        let altitude = (location.position.z ?? 0) + location.verticalAccuracy
+        
+        cameraController.originCamera = Camera(
+            latitude: location.position.y,
+            longitude: location.position.x,
+            altitude: altitude,
+            heading: calibrationHeading ?? heading,
+            pitch: 90,
+            roll: 0
+        )
+        
+        // We have to do this or the error gets bigger and bigger.
+        cameraController.transformationMatrix = .identity
+        arViewProxy.session.run(configuration, options: .resetTracking)
+        
+        // If initial camera is not set, then we set it the flag here to true
+        // and set the status text to empty.
+        if !initialCameraIsSet {
+            coachingOverlayIsActive = false
+            withAnimation {
```

Comment:
In my tests I'm not able to see the animated fade-in transition for the scene view - basemap loading takes some time. Can you see the animation on iPad? If not I think we may remove the animation.